### PR TITLE
Snyk workflow add

### PR DESCRIPTION
**Pull request by SadFaceSmith**
What this PR does / why we need it:
This PR adds a short GH Action to generate and upload SBOM reports via Snyk.

Which issue(s) this PR fixes:

Fixes #

Special notes for your reviewer:

FYSA @grafana/security-team

### CODE REVIEW

**Comment by joe-elliott** [comment]
What are SBOM reports? What is Snyk? Why do we want this change?

**Comment by SadFaceSmith** [comment]
@joe-elliott SBOM is a 'Software Bill of Materials'. Essentially a list of dependencies/libraries/etc used within a codebase/product. Snyk is a security tool that we are trialing to provide our customers with SBOM and SCA (software composition analysis) reports. 

We have multiple customers now asking for these reports from our enterprise offerings. We are integrating some of our public repos with Snyk now to get a good idea if Snyk will work across all of our codebases.

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+      - uses: actions/checkout@v3
+      - name: Run Snyk to check for vulnerabilities - sarif output
+        continue-on-error: true
+        uses: snyk/actions/golang@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: test
+          args:  --all-projects --sarif-file-output=${{ github.event.repository.name }}.sarif --strict-out-of-sync=false
```
Comment (joe-elliott) [inline .github/workflows/snyk.yml:24]: is there a reason we're running this twice with different output formats? (sarif and json)? 

is a single pass not sufficient?

File: .github/workflows/snyk.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Run Snyk to check for vulnerabilities - json output
+        continue-on-error: true
+        uses: snyk/actions/golang@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: test
+          args: --json --all-projects --json-file-output=${{ github.event.repository.name }}.json  --strict-out-of-sync=false
+
+      - name: install snyk-to-html
```
Comment (joe-elliott) [inline .github/workflows/snyk.yml:35]: i suppose a similar question here as before. any particular reason to generate all these different formats?

**Review by joe-elliott — COMMENTED** [review]
A few questions in the review. 

Also,
- Will this block PRs if it fails?
- Are the resulting reports public?

File: .github/workflows/snyk.yml (hunk 1 above)
Comment (SadFaceSmith) [reply .github/workflows/snyk.yml:24]: In my testing, you can only have one format type per run of Snyk unfortunately. We've had asks for both SARIF and JSON formats.

File: .github/workflows/snyk.yml (hunk 2 above)
Comment (SadFaceSmith) [reply .github/workflows/snyk.yml:35]: Per customer requests mostly. Some customers want the SARIF format, some SPDX, some HTML reports.

**Comment by SadFaceSmith** [comment]
@joe-elliott As of right now, yes the results are public. They are uploaded to this repo by the last Action task. They are in a zip file located on the 'Summary' page for the Action.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**